patroni/postgresql.py

```diff
@@ -1365,7 +1371,9 @@ def follow(self, member, timeout=None):
         if primary_conninfo:
             recovery_params['primary_conninfo'] = primary_conninfo
         if self.use_slots:
-            recovery_params['primary_slot_name'] = slot_name_from_member_name(self.name)
+            required_name = member is not None and member.data.get('replication_slot')
```

Comment:
In case if we are following the remote cluster and there is no `replication_slot` defined in the configuration `primary_slot_name` shouldn't be defined either.